torchvision/datasets/vision.py

```diff
@@ -0,0 +1,42 @@
+import os
+import torch.utils.data as data
+
+
+class VisionDataset(data.Dataset):
+    _repr_indent = 4
+
+    def __init__(self, root, transform, target_transform):
+        if isinstance(root, str):
```

Comment:
I wonder if this is a bit fragile in a few situations, with `unicode`, `bytes` etc?
Maybe using [`torch._six.string_classes`](https://github.com/pytorch/pytorch/blob/fbd690c1fec0651ee9e6cc07ddcb12217ffb31bc/torch/_six.py#L37-L40) would be more robust?

Reply:
My knowledge is not deep enough to answer that. On the other hand I see no drawback in implementing it that way just to be sure.